lib/token-assert.js

```diff
@@ -584,6 +584,14 @@ TokenAssert.prototype.noTrailingSpaces = function(options) {
                 fixed = true;
             }
 
+            if (!fixed && precendingToken && precendingToken.type === 'EOF') {
+                precendingToken = this._file.getPrevToken(precendingToken, { includeWhitespace: true });
+                if (precendingToken && precendingToken.isWhitespace) {
```

Comment:
I'm waiting on making this change to see if the better approach in #1981 will be considered.